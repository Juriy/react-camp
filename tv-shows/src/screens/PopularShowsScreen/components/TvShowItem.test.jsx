import TvShowItem from './TvShowItem';
import { StaticRouter as Router } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

test('renders a tv show item with a label "add to favorites"', () => {
  const TestComponent = () => (
    <Router>
      <TvShowItem />
    </Router>
  );
  render(<TestComponent />);
  // screen.debug();
  const button = screen.getByText(/add to favorites/i);
  expect(button).toHaveAttribute('type', 'button');
});

test('renders a tv show item with a label "remove from favorites" when already favorited', () => {
  const TestComponent = () => (
    <Router>
      <TvShowItem favorited />
    </Router>
  );
  render(<TestComponent />);
  //screen.debug();
  const button = screen.getByText(/remove from favorites/i);
  expect(button).toHaveAttribute('type', 'button');
});

test('calls onClick when being clicked', () => {
  const onClickMock = jest.fn(() => {});
  const TestComponent = () => (
    <Router>
      <TvShowItem onClick={onClickMock} />
    </Router>
  );
  render(<TestComponent />);

  const button = screen.getByRole('button');
  userEvent.click(button);
  expect(onClickMock).toHaveBeenCalledTimes(1);
});

test('toggle favorites button when the component is being clicked', () => {
  const onClickMock = jest.fn(() => {});
  const TestComponent = (props) => (
    <Router>
      <TvShowItem {...props} onClick={onClickMock} />
    </Router>
  );

  const { rerender } = render(<TestComponent />);

  const button = screen.getByRole('button');
  userEvent.click(button);
  expect(onClickMock).toHaveBeenCalledTimes(1);

  //rerender(<TestComponent favorited />);
});

test('matches snapshot', () => {
  const TestComponent = () => (
    <Router>
      <TvShowItem
        id={2}
        name="Game of Thrones"
        posterSrc="/path/to/poster.png"
        rating={8.5}
        overview="some overview here"
        releaseDate="11.11.2011"
      />
    </Router>
  );
  const { container } = render(<TestComponent />);
  expect(container).toMatchSnapshot();
});

test('matches snapshot for favorited', () => {
  const TestComponent = () => (
    <Router>
      <TvShowItem
        favorited
        id={2}
        name="Game of Thrones"
        posterSrc="/path/to/poster.png"
        rating={8.5}
        overview="some overview here"
        releaseDate="11.11.2011"
      />
    </Router>
  );
  const { container } = render(<TestComponent />);
  expect(container).toMatchSnapshot();
});
