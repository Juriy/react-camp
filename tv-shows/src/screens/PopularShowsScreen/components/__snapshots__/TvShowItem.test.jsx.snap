// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="ui card"
  >
    <a
      class="ui image"
      href="/details/2"
    >
      <img
        alt="Game of Thrones"
        src="https://image.tmdb.org/t/p/w500//path/to/poster.png"
        width="50"
      />
    </a>
    <div
      class="content"
    >
      <div
        class="header"
      >
        Game of Thrones
      </div>
      <div
        class="meta"
      >
        11.11.2011
      </div>
    </div>
    <div
      class="content"
    >
      <i
        aria-hidden="true"
        class="star icon"
      />
      8.5
    </div>
    <div
      class="content"
    >
      <button
        class="ui green basic fluid button"
        type="button"
      >
        Add to favorites
      </button>
    </div>
  </div>
</div>
`;

exports[`matches snapshot for favorited 1`] = `
<div>
  <div
    class="ui card"
  >
    <a
      class="ui image"
      href="/details/2"
    >
      <img
        alt="Game of Thrones"
        src="https://image.tmdb.org/t/p/w500//path/to/poster.png"
        width="50"
      />
    </a>
    <div
      class="content"
    >
      <div
        class="header"
      >
        Game of Thrones
      </div>
      <div
        class="meta"
      >
        11.11.2011
      </div>
    </div>
    <div
      class="content"
    >
      <i
        aria-hidden="true"
        class="star icon"
      />
      8.5
    </div>
    <div
      class="content"
    >
      <button
        class="ui green fluid button"
        type="button"
      >
        Remove from favorites
      </button>
    </div>
  </div>
</div>
`;
